### PR TITLE
[SQL]Fix upsert SQL template for Postgres plugin

### PR DESCRIPTION
**Pull request by longquanzheng**
**What changed?**
Fixes issue #3482 and replace https://github.com/uber/cadence/pull/3484

**Why?**

**How did you test it?**
Integration test with local Postgres.
New test is added

**Potential risks**
Replace on conflict is already broken, so no more risks.

### CODE REVIEW

File: common/persistence/persistence-tests/executionManagerTest.go [hunk 1]
```diff
@@ -1671,7 +1671,8 @@ func (s *ExecutionManagerSuite) TestDeleteWorkflow() {
 // TestDeleteCurrentWorkflow test
 func (s *ExecutionManagerSuite) TestDeleteCurrentWorkflow() {
 	if s.ExecutionManager.GetName() != "cassandra" {
-		s.T().Skip("this test is only applicable for cassandra (uses TTL based deletes)")
+		return
+		//s.T().Skip("this test is only applicable for cassandra (uses TTL based deletes)")
```
Comment (longquanzheng) [inline common/persistence/persistence-tests/executionManagerTest.go:1675]: Skip here is wrong. this will skip more tests than it thought.

**Review by chris-rcn — APPROVED** [review]
This is working in my real-world usage.

File: common/persistence/persistence-tests/executionManagerTest.go (hunk 1 above)
Comment (vancexu) [reply common/persistence/persistence-tests/executionManagerTest.go:1675]: maybe rephrase to comment to be just `"this test is only applicable for cassandra (uses TTL based deletes)"`

**Review by vancexu — COMMENTED** [review]
Do you plan to add more unit test before merge it?

**Comment by longquanzheng** [comment]
> Do you plan to add more unit test before merge it?

Yep, a new test is added.

**Review by vancexu — APPROVED** [review]
Thank you Long!
